Creating an idealised example

michaeldenes opened the pull request:
This pull-request is to include a new example notebook, using an idealised flow (the Bickley Jet) to show how `plasticparcels` can be used, without the need to download any datasets.

With #56 , this is to close #51

erikvansebille commented:
I can't comment on the ipynb file using GitHub tools, so here a list of (minor) comments:
- [ ] In cell 4, why not simply skip the VectorField plots (and use a counter that increments after every valid Field has been plotted)?
- [ ] Before cell 5, perhaps provide links to other examples where the settings dictionary _can_ be loaded?
- [ ] In cell 5, mention where the b/c constants come from? Some values seem arbitrary, otherwise?
- [ ] Perhaps for a later PR, but could the setting of fieldset-constants from the settings-dict not be a method?
- [ ] In cell 10 (and later), how would the user know it's `kernels[-2]` that needs replacing?
- [ ] In cell 29, perhaps print the final depth range with fewer digits?

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+    T = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+    S = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+
+    T[:, 0, :, :] = 25.0 # TODO: make sure the units are correct!
```
Has this been done now?

michaeldenes replied:
Yes, will remove

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+    S[:, 0, :, :] = 35.0
+
+    for d in range(1, depth.size):
+        T[:, d, :, :] = T[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
+        S[:, d, :, :] = S[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
```
WHy these commented parts of the statements? Are they stray debug messages?

michaeldenes replied:
I had a depth-varying temp/salt field, but I found it too cumbersome to test with, so I decided on a uniform in spacetime field. Will remove

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetbgc.add_periodic_halo(zonal=True)
```
Remove this commented line?

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+
+def add_wind_field(fieldset, times):
+    """ Horizontal wind field with a sinusoidal variation in time."""
+    lon = fieldset.U.grid.lon # Is this right?
```
Yes I think so; comment can be removed?

michaeldenes replied:
Thanks!

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetwind.add_periodic_halo(zonal=True)
```
Remove commented message?

michaeldenes replied:
Done

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetStokes.add_periodic_halo(zonal=True)
```
Remove commented message?

michaeldenes replied:
Done

erikvansebille commented on docs/examples/idealised_flow.py:
```diff
@@ -0,0 +1,261 @@
+
+def add_stokes_field(fieldset, times):
+    """ Horizontal wave field that varies in time."""
+    lon = fieldset.U.grid.lon # Is this right?
```
Remove message?

michaeldenes replied:
Done

michaeldenes commented:
> * [ ]  Before cell 5, perhaps provide links to other examples where the settings dictionary _can_ be loaded?

I've made a link to the examples page, just using 'see other examples'. Or should we flesh this out more?
> * [ ]  In cell 5, mention where the b/c constants come from? Some values seem arbitrary, otherwise?

I've added to the comment that they come from the Kooi et al, Lobelle et al, and Kaandorp et al papers, with dois.
> * [ ]  Perhaps for a later PR, but could the setting of fieldset-constants from the settings-dict not be a method?

I'm not sure I understand this one just yet, but maybe there's some refactoring we can do, yep!
> * [ ]  In cell 10 (and later), how would the user know it's `kernels[-2]` that needs replacing?

I've added a note for cell 10, but have left all the later ones as is. Or should we add the note to each?
> * [ ]  In cell 29, perhaps print the final depth range with fewer digits?

I've made it 4dp now. Perhaps we also print the difference?

erikvansebille commented:
Thanks for incorporating these, @michaeldenes. Two more comments on the notebook
1. The new example is not listed at https://plasticparcels--55.org.readthedocs.build/en/55/examples.html?
2. The else statement in cell 4 can be removed (instead of commented out)?

michaeldenes commented:
> Thanks for incorporating these, @michaeldenes. Two more comments on the notebook
> 
> 1. The new example is not listed at https://plasticparcels--55.org.readthedocs.build/en/55/examples.html?
> 2. The else statement in cell 4 can be removed (instead of commented out)?

Thanks @erikvansebille, I've now made these two changes
